pkg/httpconfig/http.go

```diff
@@ -335,7 +337,11 @@ func NewClient(logger log.Logger, cfg EndpointsConfig, client *http.Client, prov
 		if err != nil {
 			return nil, err
 		}
-		discoverers = append(discoverers, file.NewDiscovery(&fileSDCfg, logger))
+		discovery, err := file.NewDiscovery(&fileSDCfg, logger, reg)
```

Comment:
It will register the SD metrics multiple times and cause it to fail.
Should we wrap the registry somehow?

Reply:
I am ok either way. Does the current way work? Not sure if we have tests to capture the issue

Reply:
Since these configs don't have a name, I am not sure if we could prefix the registry with anything meaningful other than the ordinal index. For now maybe we can ignore these metrics and we can see how to include them if someone needs them.

Reply:
Makes sense to me